app/v1/services/RetrieveForeignReliefsService.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.services
+
+import cats.data.EitherT
+import cats.implicits._
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.Logging
+import v1.connectors.RetrieveForeignReliefsConnector
+import v1.controllers.EndpointLogContext
+import v1.models.errors._
+import v1.models.request.retrieveForeignReliefs.RetrieveForeignReliefsRequest
+import v1.support.DesResponseMappingSupport
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RetrieveForeignReliefsService @Inject()(connector: RetrieveForeignReliefsConnector) extends DesResponseMappingSupport with Logging {
+
+  def retrieve(request: RetrieveForeignReliefsRequest)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext,
+    logContext: EndpointLogContext): Future[RetrieveForeignReliefsServiceOutcome] = {
+
+    val result = for {
+      desResponseWrapper <- EitherT(connector.retrieve(request)).leftMap(mapDesErrors(desErrorMap))
+    } yield desResponseWrapper
+
+    result.value
+  }
+
+  private def desErrorMap =
+    Map(
+      "INVALID_TAXABLE_ENTITY_ID" -> NinoFormatError,
+      "FORMAT_TAX_YEAR" -> TaxYearFormatError,
+      "CLIENT_OR_AGENT_NOT_AUTHORISED" -> UnauthorisedError,
```

Comment:
CLIENT_OR_AGENT_NO_AUTHORISED error shouldn't be in des error mapping as it's a frontend error.

I'm aware that some of the other services also include it, discussing with James about those